api/handler/v1beta1/appeal.go

```diff
@@ -106,6 +106,10 @@ func (s *GRPCServer) CreateAppeal(ctx context.Context, req *guardianv1beta1.Crea
 
 func (s *GRPCServer) GetAppeal(ctx context.Context, req *guardianv1beta1.GetAppealRequest) (*guardianv1beta1.GetAppealResponse, error) {
 	id := req.GetId()
+	if !utils.IsValidUUID(id) {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid appeal id: %v", id)
```

Comment:
let's do this in the service layer instead?

Reply:
done